### ISSUE TITLE
Unclear  warning in aggregated_by docstring.

### ISSUE THREAD

**Issue by ajdawson** [SciTools/iris#1758]
The docstring for `iris.cube.Cube.aggregated_by` contains this:

```
TODO: It is not clear if repeating values must be consecutive to form a group.
```

This looks like this has been in there since the initial commit into git... Should it still be there? It doesn't strike me as being relevant, I think the aggregation works fine with non-consecutive groups doesn't it? Anyway, it is rather off-putting and at very least could do with clarification so users know what they can expect!

**Comment by pelson** [comment]
Completely agree. Looks like somebody added this after the actual implementation. It should be easy to determine the facts. @bjlittle - do you have any recollection of aggregated_by? I seem to remember you writing some/all of it, so suspect there is even a test for this case :wink:

**Comment by ajdawson** [comment]
Closed by #2008
